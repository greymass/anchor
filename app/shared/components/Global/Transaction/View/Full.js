// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import ReactJson from 'react-json-view';

import { Header } from 'semantic-ui-react';

type Props = {
  transaction: {}
};

class GlobalTransactionViewFull extends Component<Props> {
  props: Props;
  render() {
    const {
      t,
      transaction
    } = this.props;
    return (
      <React.Fragment>
        <Header>
          {t('global_transaction_full_details_title')}
          <Header.Subheader>
            {t('global_transaction_full_details_content')}
          </Header.Subheader>
        </Header>
        <ReactJson
          displayDataTypes={false}
          displayObjectSize={false}
          iconStyle="square"
          name={null}
          src={transaction}
          style={{ padding: '1em' }}
          theme="harmonic"
        />
      </React.Fragment>
    );
  }
}

export default translate('global')(GlobalTransactionViewFull);
