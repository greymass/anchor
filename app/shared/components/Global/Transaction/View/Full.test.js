import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalTransactionViewFull } from './Full';

const mockProps = {
  t: text => text,
  transaction: {},
};

describe('GlobalTransactionViewFull', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalTransactionViewFull {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

