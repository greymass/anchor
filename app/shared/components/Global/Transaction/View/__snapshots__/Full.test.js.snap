// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTransactionViewFull renders correctly 1`] = `
<Fragment>
  <Header>
    global_transaction_full_details_title
    <HeaderSubheader>
      global_transaction_full_details_content
    </HeaderSubheader>
  </Header>
  <t
    collapseStringsAfterLength={32}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={false}
    displayObjectSize={false}
    enableClipboard={true}
    groupArraysAfterLength={100}
    iconStyle="square"
    indentWidth={4}
    name={null}
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={Object {}}
    style={
      Object {
        "padding": "1em",
      }
    }
    theme="harmonic"
    validationMessage="Validation Error"
  />
</Fragment>
`;
